components/availability/src/Availability.svelte

```diff
@@ -171,20 +177,81 @@
     }
   };
 
+  // Consecutive same-availability periods of time, from earliest start_time to latest end_time.
+  const generateEpochs = (
+    slots: SelectableSlot[],
+    participant_threshold: number,
+  ) => {
+    let scale = scaleTime().domain([
+      slots[0].start_time,
+      slots[slots.length - 1].end_time,
+    ]);
+    let epochs = slots
+      .reduce((m, n) => {
+        if (
+          m[m.length - 1] &&
+          JSON.stringify(m[m.length - 1][0].available_calendars) ===
+            JSON.stringify(n.available_calendars)
+        ) {
+          m[m.length - 1].push(n);
```

Comment:
Is this to use `push()` to set `n` as the second last element of `m`?

Reply:
Sort of -- although "second last" is tenuous, since the reduce is still running at assignment time.

This push says "Add me to the last epoch, instead of creating a new epoch" -- @ozsivanov gave some great suggestions on clarifying the language here which I'll be pushing up shortly. Thanks!